usaon_vta_survey/util/full_sankey.py

```diff
@@ -0,0 +1,97 @@
+from itertools import chain
+
+from usaon_vta_survey.models.tables import (
+    Response,
+    ResponseApplication,
+    ResponseDataProduct,
+    ResponseObservingSystem,
+)
+
+
+# TODO: Can we do better than `object` here? Mypy doesn't correctly infer `str | int`
+def sankey(response: Response) -> list[list[object]]:
+    """Provide Sankey data structure, formatted for Highcharts."""
+    # Convert tuples to lists for passing to Javascript-land:
+    data = [list(e) for e in _sankey(response)]
+    return data
+
+
+def _sankey(response: Response) -> list[tuple[str, str, int]]:
+    """Provide a sankey data structure of applications, formatted for type checker."""
+    data1 = _applications_sankey(response)
+    data2 = _data_products_sankey(response)
+    data3 = _societal_benefit_areas_sankey(response)
+    data = data1 + data2 + data3
+    return data
+
+
+def _societal_benefit_areas_sankey(response: Response):
+    data = list(
+        chain(
+            *[
+                _application_societal_benefit_area_sankey_links(application)
+                for application in response.applications
+            ]
+        )
+    )
+    return data
+
+
+def _applications_sankey(response: Response) -> list[tuple[str, str, int]]:
+    """Provide a sankey data structure of applications, formatted for type checker."""
```

Comment:
>, formatted for type checker

What do you mean by this?

Reply:
This was copied over from the prototype. Probably could be deleted.

Reply:
Wow, I have no idea what I meant by that :laughing: